### PR TITLE
Unprompted: add public key to context and header

### LINKED ISSUE #2599: Add public key to context
Based on discussions with @jhoyla, @chris-wood, and others, there are potential key confusion attacks when the server is checking the signature using the wrong public key. I think the simplest solution is to add the public key to the TLS key exporter. That should solve the issue without increasing the amount of bytes sent on the wire

### PR DESCRIPTION
**Pull request by DavidSchinazi**
Fixes #2599

### CODE REVIEW

File: draft-ietf-httpbis-unprompted-auth.md [hunk 1]
```diff
@@ -201,6 +203,11 @@ Key ID:
 
 : The key ID sent in the `k` Parameter (see {{parameter-k}}).
 
+Public Key:
+
+: The public key used by the server to validate the signature provided by the
+client.
```
Comment (chris-wood) [inline draft-ietf-httpbis-unprompted-auth.md:209]: What is the encoding of this public key? We probably need to be clear about that in order for this to work.

**Review by chris-wood — CHANGES_REQUESTED** [review]
Looks good! The only thing I suggest we include are the specifics of how the public key is encoded.

File: draft-ietf-httpbis-unprompted-auth.md (hunk 1 above)
Comment (DavidSchinazi) [reply draft-ietf-httpbis-unprompted-auth.md]: Good point. Fixed in 9dc88eef6543e47dd32c84427a1df73424e4edff

File: draft-ietf-httpbis-unprompted-auth.md [hunk 2]
```diff
@@ -304,6 +340,13 @@ The REQUIRED "k" (key ID) parameter is a byte sequence that identifies which key
 the user agent wishes to use to authenticate. This can for example be used to
 point to an entry into a server-side database of known keys.
 
+## The a Parameter {#parameter-a}
+
+The REQUIRED "a" (public key) parameter is a byte sequence that contains the
+public key used by the server to validate the signature provided by the client.
+This avoids key confusion issues (see {{SEEMS-LEGIT}}). The encoding of the
+public key is described in {{context}}.
```
Comment (chris-wood) [inline draft-ietf-httpbis-unprompted-auth.md:348]: Should we add text which describes what to do if the encoding of this parameter is invalid, or is that already covered elsewhere?

**Review by chris-wood — APPROVED** [review]

File: draft-ietf-httpbis-unprompted-auth.md (hunk 2 above)
Comment (DavidSchinazi) [reply draft-ietf-httpbis-unprompted-auth.md:348]: I think that's covered by the `mismatch between key ID and provided public key` text (you don't need to check for invalid encoding, just `memcmp` with your known valid encoding), but this is making me want to make that normative. Will do in followup PR.

Comment (chris-wood) [reply draft-ietf-httpbis-unprompted-auth.md:348]: Sounds good!

**Pull request merged**